src/compiler/ast/module/SourceFile.ts

```diff
@@ -744,6 +744,40 @@ export class SourceFile extends SourceFileBase<ts.SourceFile> {
         return this;
     }
 
+    /**
+     * Removes all unused declarations like interfaces, classes, enums, functions, variables, parameters,
+     * methods, properties, imports, etc. from this file.
+     *
+     * WARNING! This will forget all the nodes in the file! It's best to do this after you're all done with the file.
+     */
+    removeUnusedDeclarations(formatSettings: FormatCodeSettings = {}, userPreferences: UserPreferences = {}) {
+        const sourceFile = this;
+        // heads up: calling it twice removes more unused variables, like things referenced in unused things or named variables
+        applyCombinedCodeFix(this._context.languageService.getCombinedCodeFix(this, "unusedIdentifier_delete", formatSettings, userPreferences));
+        applyCombinedCodeFix(this._context.languageService.getCombinedCodeFix(this, "unusedIdentifier_delete", formatSettings, userPreferences));
```

Comment:
Use:

```ts
this._context.languageService.getCombinedCodeFix(this, "unusedIdentifier_delete", formatSettings, userPreferences).applyChanges();
```

Don't worry about how that will forget all the nodes. We can improve it in the future if necessary.